<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TwitFil</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QVBoxLayout" name="mainLayout">
      <property name="spacing">
       <number>-1</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="patternLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <layout class="QVBoxLayout" name="inactivePatternLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="inactivePatternsLabel">
            <property name="font">
             <font>
              <pointsize>18</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Inactive Patterns</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="inactivePatternsWidget"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="patternButtonsLayout">
          <property name="spacing">
           <number>-1</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QPushButton" name="deactivatePatternButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>&lt;-</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="activatePatternButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>-&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="activePatternLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="activePatternsLabel">
            <property name="font">
             <font>
              <pointsize>18</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Active Patterns</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="activePatternsWidget"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="managePatternsLayout">
        <item>
         <widget class="QPushButton" name="addPatternButton">
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="editPatternButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Edit</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deletePatternButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="mainControlLayout">
        <item>
         <widget class="QPushButton" name="falsePatternButton">
          <property name="text">
           <string>Record False Patterns</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="toggleRecognizerButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Launch Recognizer</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="micStatusWidget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>15</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>0</width>
          <height>15</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Preferred</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="micInputLabel">
        <property name="font">
         <font>
          <pointsize>18</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Microphone Input</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="IODevicePlotterWidget" name="micInputGraph"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuTwitFil">
    <property name="title">
     <string>TwitFil</string>
    </property>
   </widget>
   <addaction name="menuTwitFil"/>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>IODevicePlotterWidget</class>
   <extends>QGraphicsView</extends>
   <header location="global">iodeviceplotterwidget.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>addPatternButton</tabstop>
  <tabstop>editPatternButton</tabstop>
  <tabstop>deletePatternButton</tabstop>
  <tabstop>activatePatternButton</tabstop>
  <tabstop>deactivatePatternButton</tabstop>
  <tabstop>inactivePatternsWidget</tabstop>
  <tabstop>activePatternsWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>MainWindow</sender>
   <signal>editButtonEnabledChanged(bool)</signal>
   <receiver>deletePatternButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>791</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>editButtonEnabledChanged(bool)</signal>
   <receiver>editPatternButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>657</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inactivePatternsWidget</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>activePatternsWidget</receiver>
   <slot>clearSelection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>activePatternsWidget</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>inactivePatternsWidget</receiver>
   <slot>clearSelection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>toActiveButtonEnabledChanged(bool)</signal>
   <receiver>activatePatternButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>661</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>toInactiveButtonEnabledChanged(bool)</signal>
   <receiver>deactivatePatternButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>370</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>385</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>recognizerButtonEnabledChanged(bool)</signal>
   <receiver>toggleRecognizerButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>toInactiveButtonEnabledChanged(bool)</signal>
  <signal>toActiveButtonEnabledChanged(bool)</signal>
  <signal>editButtonEnabledChanged(bool)</signal>
  <signal>recognizerButtonEnabledChanged(bool)</signal>
 </slots>
</ui>
